lib/Markdown/Perl/BlockParser.pm

```diff
@@ -374,6 +374,9 @@ sub _parse_yaml_metadata {
       pos($this->{md}) = 0;
```

Comment:
Maybe there should be a call to `carp` in this branch, to make such an error be easier to discover.